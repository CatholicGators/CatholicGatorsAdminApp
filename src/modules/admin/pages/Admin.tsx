import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  Theme,
  createStyles,
  withStyles,
  Paper,
  Table,
  TableHead,
  TableRow,
  TableCell,
  TableBody,
  Avatar,
  Switch,
  Toolbar,
  Typography,
  CircularProgress
} from '@material-ui/core';

import {
  getUsers,
  updateUser
} from '../../../redux/actions/admin/adminActions';

const styles = (theme: Theme) => createStyles({
  tableWrapper: {
    maxWidth: '1000px',
    marginTop: '50px',
    marginLeft: 'auto',
    marginRight: 'auto'
  },
  tableCard: {
    overflowX: 'auto'
  },
  tableLoadingContainer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    minHeight: '300px',
  }
})

type Props = {
  classes: any
  users: any
  getUsers: () => any,
  updateUser: (user) => any
}

export class Admin extends Component<Props> {
  componentDidMount() {
    this.props.getUsers()
  }

  handleApprovedToggle(user, checked) {
    this.props.updateUser({
      ...user,
      data: {
        ...user.data,
        isApproved: checked
      }
    })
  }

  handleAdminToggle(user, checked) {
    this.props.updateUser({
      ...user,
      data: {
        ...user.data,
        isAdmin: checked
      }
    })
  }

  render() {
    const { classes, users } = this.props;

    return (
      <div className={classes ? classes.tableWrapper : null}>
        <Paper className={classes ? classes.tableCard : null}>
          <Toolbar>
            <Typography variant="h6">
              Users
            </Typography>
          </Toolbar>
          <Table>
            <TableHead>
              <TableRow>
                <TableCell>Profile Pic</TableCell>
                <TableCell>Name</TableCell>
                <TableCell>Email</TableCell>
                <TableCell>Approved</TableCell>
                <TableCell>Admin</TableCell>
              </TableRow>
            </TableHead>
            {users ? 
              <TableBody>
                {users.map(user => (
                  <TableRow key={user.id}>
                    <TableCell>
                      <Avatar src={user.photoURL} />
                    </TableCell>
                    <TableCell>{user.name}</TableCell>
                    <TableCell>{user.email}</TableCell>
                    <TableCell>
                      <Switch
                        checked={user.isApproved}
                        onChange={(_, checked) => this.handleApprovedToggle(user, checked)}/>
                    </TableCell>
                    <TableCell>
                      <Switch
                        disabled={!user.isApproved}
                        checked={user.isAdmin}
                        onChange={(_, checked) => this.handleAdminToggle(user, checked)}
                      />
                    </TableCell>
                  </TableRow>
                ))}
              </TableBody>
            : null }
          </Table>
          {!users ?
            <div className={classes ? classes.tableLoadingContainer : null}>
              <CircularProgress />
            </div>
          : null}
        </Paper>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  users: state.admin.users
})

const mapDispatchToProps = dispatch => ({
  getUsers: () => dispatch(getUsers()),
  updateUser: user => dispatch(updateUser(user))
})

export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(Admin))
