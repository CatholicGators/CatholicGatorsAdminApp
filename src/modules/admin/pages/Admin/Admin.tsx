import React, { Component } from "react"
import { Route, Redirect, NavLink } from "react-router-dom"
import {
    Theme,
    createStyles,
    withStyles,
    ListItem,
    ListItemText,
    List,
    ListItemIcon,
} from "@material-ui/core"
import { VpnKey, Group } from "@material-ui/icons"
import UserTable from "../../modules/users/components/UserTable/UserTable"
import Interests from "../../modules/interests/components/Interests/Interests"

export const styles = (theme: Theme) =>
    createStyles({
        pageWrapper: {
            display: "flex",
            margin: "50px 0px",
            padding: `0px ${theme.spacing(1)}px`,
            width: "100%",
        },
        pageNav: {
            width: 250,
            minWidth: 250,
            padding: 0,
            marginRight: `${theme.spacing(1)}px`,
        },
        navLink: {
            textDecoration: "none",
            "&:visited": {
                color: "inherit",
            },
        },
        colorInitial: {
            color: "initial",
        },
    })

type Props = {
    classes: any
    match: any
    location: any
}

export type MenuLink = {
    text: string
    href: string
    icon: any
}
export const menuLinks: Array<MenuLink> = [
    {
        text: "Users",
        href: "/users",
        icon: Group,
    },
    {
        text: "Interests",
        href: "/interests",
        icon: VpnKey,
    },
]

export class Admin extends Component<Props> {
    render() {
        const { classes, match, location } = this.props

        return (
            <div className={classes.pageWrapper}>
                <List className={classes.pageNav}>
                    {menuLinks.map((link) => (
                        <NavLink
                            exact
                            key={link.text}
                            to={`${match.url}${link.href}`}
                            className={classes.navLink}
                        >
                            <ListItem
                                button
                                className={classes.colorInitial}
                                selected={
                                    location.pathname ===
                                    `${match.url}${link.href}`
                                }
                            >
                                <ListItemIcon>
                                    <link.icon />
                                </ListItemIcon>
                                <ListItemText primary={link.text} />
                            </ListItem>
                        </NavLink>
                    ))}
                </List>
                <Route
                    exact
                    path={`${match.url}`}
                    render={() => <Redirect to={`${match.url}/users`} />}
                />
                <Route path={`${match.url}/users`} component={UserTable} />
                <Route path={`${match.url}/interests`} component={Interests} />
            </div>
        )
    }
}

export default withStyles(styles)(Admin)
