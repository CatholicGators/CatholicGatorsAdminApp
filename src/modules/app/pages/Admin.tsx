import React, { Component } from 'react';
import { listenForUser } from '../../../redux/actions/auth/authActions';
import { connect } from 'react-redux';

type Props = {
  listenForUser: () => any;
}

export class Admin extends Component<Props> {
  render() {
    return (
      <div>
        <h1>Admin!</h1>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  listenForUser: () => dispatch(listenForUser())
});

export default connect(null, mapDispatchToProps)(Admin);
