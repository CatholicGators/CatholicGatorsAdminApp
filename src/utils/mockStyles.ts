import { createMuiTheme } from '@material-ui/core'

export default (styles) => Object.keys(styles(createMuiTheme())).reduce((acc, curr) => ({...acc, [curr]: "test"}), {})