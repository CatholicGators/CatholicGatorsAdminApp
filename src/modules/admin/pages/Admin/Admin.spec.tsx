import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { Admin, menuLinks, styles } from './Admin'
import mockStyles from '../../../../utils/mockStyles'

describe('Admin', () => {
    let props, wrapper

    beforeEach(() => {
        props = {
            classes: mockStyles(styles),
            match: {
                url: 'test'
            },
            location: {
                pathname: 'test/testing'
            }
        }
        wrapper = shallow(<Admin {...props} />)
    })

    it('should match snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('should render all of the sub nav links', () => {
        menuLinks.forEach(link =>
            expect(
                wrapper.exists(`NavLink[to='${props.match.url}${link.href}']`)
            ).toBe(true)
        )
    })
})
