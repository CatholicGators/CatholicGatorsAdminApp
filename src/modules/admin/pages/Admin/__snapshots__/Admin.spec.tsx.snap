// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Admin
    location={
      Object {
        "pathname": "test/testing",
      }
    }
    match="test/"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(List)
          className={null}
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            className={null}
            exact={true}
            to="undefined/users"
          >
            <WithStyles(ListItem)
              button={true}
              selected={false}
            >
              <WithStyles(ListItemIcon)>
                <pure(GroupIcon) />
              </WithStyles(ListItemIcon)>
              <WithStyles(ListItemText)
                primary="Users"
              />
            </WithStyles(ListItem)>
          </NavLink>
          <NavLink
            activeClassName="active"
            aria-current="page"
            className={null}
            exact={true}
            to="undefined/permissions"
          >
            <WithStyles(ListItem)
              button={true}
              selected={false}
            >
              <WithStyles(ListItemIcon)>
                <pure(VpnKeyIcon) />
              </WithStyles(ListItemIcon)>
              <WithStyles(ListItemText)
                primary="Permissions"
              />
            </WithStyles(ListItem)>
          </NavLink>
        </WithStyles(List)>,
        <Route
          exact={true}
          path="undefined"
          render={[Function]}
        />,
        <Route
          component={[Function]}
          path="undefined/users"
        />,
        <Route
          component={[Function]}
          path="undefined/permissions"
        />,
      ],
      "className": null,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              aria-current="page"
              className={null}
              exact={true}
              to="undefined/users"
            >
              <WithStyles(ListItem)
                button={true}
                selected={false}
              >
                <WithStyles(ListItemIcon)>
                  <pure(GroupIcon) />
                </WithStyles(ListItemIcon)>
                <WithStyles(ListItemText)
                  primary="Users"
                />
              </WithStyles(ListItem)>
            </NavLink>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              className={null}
              exact={true}
              to="undefined/permissions"
            >
              <WithStyles(ListItem)
                button={true}
                selected={false}
              >
                <WithStyles(ListItemIcon)>
                  <pure(VpnKeyIcon) />
                </WithStyles(ListItemIcon)>
                <WithStyles(ListItemText)
                  primary="Permissions"
                />
              </WithStyles(ListItem)>
            </NavLink>,
          ],
          "className": null,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Users",
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": <WithStyles(ListItem)
                button={true}
                selected={false}
              >
                <WithStyles(ListItemIcon)>
                  <pure(GroupIcon) />
                </WithStyles(ListItemIcon)>
                <WithStyles(ListItemText)
                  primary="Users"
                />
              </WithStyles(ListItem)>,
              "className": null,
              "exact": true,
              "to": "undefined/users",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "button": true,
                "children": Array [
                  <WithStyles(ListItemIcon)>
                    <pure(GroupIcon) />
                  </WithStyles(ListItemIcon)>,
                  <WithStyles(ListItemText)
                    primary="Users"
                  />,
                ],
                "selected": false,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <pure(GroupIcon) />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "primary": "Users",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Permissions",
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": <WithStyles(ListItem)
                button={true}
                selected={false}
              >
                <WithStyles(ListItemIcon)>
                  <pure(VpnKeyIcon) />
                </WithStyles(ListItemIcon)>
                <WithStyles(ListItemText)
                  primary="Permissions"
                />
              </WithStyles(ListItem)>,
              "className": null,
              "exact": true,
              "to": "undefined/permissions",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "button": true,
                "children": Array [
                  <WithStyles(ListItemIcon)>
                    <pure(VpnKeyIcon) />
                  </WithStyles(ListItemIcon)>,
                  <WithStyles(ListItemText)
                    primary="Permissions"
                  />,
                ],
                "selected": false,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <pure(VpnKeyIcon) />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "primary": "Permissions",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "undefined",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "undefined/users",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "undefined/permissions",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(List)
            className={null}
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              className={null}
              exact={true}
              to="undefined/users"
            >
              <WithStyles(ListItem)
                button={true}
                selected={false}
              >
                <WithStyles(ListItemIcon)>
                  <pure(GroupIcon) />
                </WithStyles(ListItemIcon)>
                <WithStyles(ListItemText)
                  primary="Users"
                />
              </WithStyles(ListItem)>
            </NavLink>
            <NavLink
              activeClassName="active"
              aria-current="page"
              className={null}
              exact={true}
              to="undefined/permissions"
            >
              <WithStyles(ListItem)
                button={true}
                selected={false}
              >
                <WithStyles(ListItemIcon)>
                  <pure(VpnKeyIcon) />
                </WithStyles(ListItemIcon)>
                <WithStyles(ListItemText)
                  primary="Permissions"
                />
              </WithStyles(ListItem)>
            </NavLink>
          </WithStyles(List)>,
          <Route
            exact={true}
            path="undefined"
            render={[Function]}
          />,
          <Route
            component={[Function]}
            path="undefined/users"
          />,
          <Route
            component={[Function]}
            path="undefined/permissions"
          />,
        ],
        "className": null,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                aria-current="page"
                className={null}
                exact={true}
                to="undefined/users"
              >
                <WithStyles(ListItem)
                  button={true}
                  selected={false}
                >
                  <WithStyles(ListItemIcon)>
                    <pure(GroupIcon) />
                  </WithStyles(ListItemIcon)>
                  <WithStyles(ListItemText)
                    primary="Users"
                  />
                </WithStyles(ListItem)>
              </NavLink>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                className={null}
                exact={true}
                to="undefined/permissions"
              >
                <WithStyles(ListItem)
                  button={true}
                  selected={false}
                >
                  <WithStyles(ListItemIcon)>
                    <pure(VpnKeyIcon) />
                  </WithStyles(ListItemIcon)>
                  <WithStyles(ListItemText)
                    primary="Permissions"
                  />
                </WithStyles(ListItem)>
              </NavLink>,
            ],
            "className": null,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Users",
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": <WithStyles(ListItem)
                  button={true}
                  selected={false}
                >
                  <WithStyles(ListItemIcon)>
                    <pure(GroupIcon) />
                  </WithStyles(ListItemIcon)>
                  <WithStyles(ListItemText)
                    primary="Users"
                  />
                </WithStyles(ListItem)>,
                "className": null,
                "exact": true,
                "to": "undefined/users",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "button": true,
                  "children": Array [
                    <WithStyles(ListItemIcon)>
                      <pure(GroupIcon) />
                    </WithStyles(ListItemIcon)>,
                    <WithStyles(ListItemText)
                      primary="Users"
                    />,
                  ],
                  "selected": false,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <pure(GroupIcon) />,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "primary": "Users",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Permissions",
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": <WithStyles(ListItem)
                  button={true}
                  selected={false}
                >
                  <WithStyles(ListItemIcon)>
                    <pure(VpnKeyIcon) />
                  </WithStyles(ListItemIcon)>
                  <WithStyles(ListItemText)
                    primary="Permissions"
                  />
                </WithStyles(ListItem)>,
                "className": null,
                "exact": true,
                "to": "undefined/permissions",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "button": true,
                  "children": Array [
                    <WithStyles(ListItemIcon)>
                      <pure(VpnKeyIcon) />
                    </WithStyles(ListItemIcon)>,
                    <WithStyles(ListItemText)
                      primary="Permissions"
                    />,
                  ],
                  "selected": false,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <pure(VpnKeyIcon) />,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "primary": "Permissions",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "undefined",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "undefined/users",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "undefined/permissions",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
